deepmd/pt/utils/dataloader.py

```diff
@@ -218,11 +218,13 @@ def __init__(self, queue, source) -> None:
         self._source = source  # Main DL iterator
 
     def run(self) -> None:
-        for item in self._source:
-            self._queue.put(item)  # Blocking if the queue is full
-
-        # Signal the consumer we are done; this should not happen for DataLoader
-        self._queue.put(StopIteration())
+        try:
+            for item in self._source:
+                self._queue.put(item)  # Blocking if the queue is full
+        except Exception as e:
+            self._queue.put(e)
```

Comment:
Note that the exception is not covered by ut. And what possible exception will be raised instead of `StopIteration` here?

Reply:
In case of the pytorch dataloader throws unexpected exceptions. Generally this code path is not used.